src/transformers/models/regnet/modeling_regnet.py

```diff
@@ -0,0 +1,486 @@
+    "zuppif/regnety-040",
+    # See all regnet models at https://huggingface.co/models?filter=regnet
+]
+
+
+@dataclass
+# Copied from transformers.models.resnet.modeling_resnet.ResNetEncoderOutput with ResNet->RegNet
+class RegNetEncoderOutput(ModelOutput):
+    """
+    RegNet encoder's output, with potential hidden states.
+
+    Args:
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings + one for the output of each layer) of
+            shape `(batch_size, num_channels, height, width)`.
+
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+    """
+
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+# Copied from transformers.models.resnet.modeling_resnet.ResNetModelOutput with ResNet->RegNet
+class RegNetModelOutput(ModelOutput):
+    """
+    RegNet model's output, with potential hidden states.
+
+    Args:
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        pooler_output (`torch.FloatTensor` of shape `(batch_size, config.hidden_sizes[-1])`):
+           The pooled last hidden state.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings + one for the output of each layer) of
+            shape `(batch_size, num_channels, height, width)`.
+
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+    """
+
+    last_hidden_state: torch.FloatTensor = None
+    pooler_output: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class RegNetConvLayer(nn.Sequential):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int = 3,
+        stride: int = 1,
+        activation: str = "relu",
+        groups: int = 1,
+    ):
+        super().__init__()
+        self.conv = nn.Conv2d(
+            in_channels,
+            out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=kernel_size // 2,
+            groups=groups,
+            bias=False,
+        )
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.act = ACT2FN[activation] if activation is not None else nn.Identity()
+
+
+class RegNetEmbeddings(nn.Sequential):
```

Comment:
Not sure if we want to use `nn.Sequential` as it's harder to debug.

cc @sgugger

Reply:
I don't agree, `Sequential` is used in very basic cases. You can either add a forward if you want to debug (but really?) or you place a break point where before/after layer is used. 

If for some reason the code fails inside the `forward` in the `sequential` the debug will stop the execution there, so no problem

Reply:
I just pinged Lysandre and Patrick on this specific one. @NielsRogge makes a good argument and we should avoid subclassing `nn.Sequential` for easier debugging (printing shapes inside or adding a break point). As you say, it's not a very complicated module, so writing the forward pass should be easy enough.

Reply:
I don't agree with the suggestions but I am happy to remove the `nn.Sequential` superclasses to align with our convention

Reply:
Update all sequentials